mpr-research-data.py

```diff
@@ -144,51 +107,164 @@ def sliceAndPushToGCP(courseDF, retrieveDF, client, targetBucketName=targetBucke
             logging.error(
                 f'Failed to upload Course Data for {outputFilename} to GCP.')
+            allSaved = False
             continue
 
-    return True
-
-
-# RETRIEVE KEYS
-# --------------------------------------------------------------------------
-
-dbParams = getDBCreds()
-gcpParams = getGCPCreds()
-
-
-# ESTABLISH CONNECTIONS
-# --------------------------------------------------------------------------
-sqlEngine = makeDBConnection(dbParams)
-gcpClient = makeGCPConnection(gcpParams)
-# sys.stdout.flush()
-
-# RETRIEVE COURSE INFO
-# --------------------------------------------------------------------------
-courseQueryDF = courseQueryMaker(
-    queryTemplateDict['course'], numberOfMonths, sqlEngine)
-# sys.stdout.flush()
-
-# RETRIEVE COURSE DATA
-# --------------------------------------------------------------------------
-retrieveQueryDF = retrieveQueryMaker(
-    queryTemplateDict['retrieve'], courseQueryDF['id'], sqlEngine)
-# sys.stdout.flush()
-
-# SEND TO GCP BUCKET
-# --------------------------------------------------------------------------
-sliceAndPushToGCP(courseQueryDF, retrieveQueryDF, gcpClient, targetBucketName)
-# sys.stdout.flush()
+    return allSliced, allSaved
+
+
+class Config:
+    def __init__(self):
+        self.logLevel: str = 'INFO'
+        self.targetBucketName: str = 'mpr-research-data-uploads'
+        self.numberOfMonths: int = 4
+        self.defaultQueryFolder: str = 'queries'
+        self.queryTemplateDict: str = {'course': 'courseQuery.sql',
+                                       'retrieve': 'retrieveQuery.sql'}
+        self.dbParams: dict = {'NAME': None,
+                               'USER': None,
+                               'PASSWORD': None,
+                               'HOST': None,
+                               'PORT': 3306}
+        self.gcpParams: dict = {}
+
+    def set(self, name, value):
+        if name in self.__dict__:
+            self.name = value
+        else:
+            raise NameError('Name not accepted in set() method')
+
+    def configFetch(self, name, default=None, casting=None, validation=None, valErrorMsg=None):
+        value = os.getenv(name, default)
+        if (casting is not None):
+            try:
+                value = casting(value)
+            except:
+                errorMsg = f'Casting error for config item "{name}" value "{value}".'
+                logging.error(errorMsg)
+                return None
+                #raise TypeError(errorMsg)
+        if (validation is not None and not validation(value)):
+            errorMsg = f'Validation error for config item "{name}" value "{value}".'
+            logging.error(errorMsg)
+            return None
+            #raise ValueError(errorMsg)
+        return value
+
+    def setFromEnv(self):
 
-logging.info('All steps complete.')
+        try:
+            self.logLevel = str(os.environ.get('LOG_LEVEL', self.logLevel)).upper()
```

Comment:
Here, `os.environ.get()` is used.  In `configFetch()`, `os.getenv()` is used (line 153).  There are minor differences between the two functions.  One suggestion I saw was that `os.getenv()` **_shouldn't_** be used only because its corresponding `os.putenv()` is broken.  So it's better to use the functions in the `os.environ` module to avoid mistakes.

Look it up and decide for yourself.  Either way, we should use the same function on both lines for getting the values of the environment variables.

Reply:
Will fix that to `os.environ.get()` everywhere.

Reply:
Sounds good to me!  ✅